modules/effects/src/map_to_action.ts

```diff
@@ -0,0 +1,168 @@
+import { Action } from '@ngrx/store';
+import {
+  defer,
+  merge,
+  Notification,
+  Observable,
+  OperatorFunction,
+  Subject,
+  NotificationKind,
+} from 'rxjs';
+import {
+  concatMap,
+  dematerialize,
+  filter,
+  finalize,
+  map,
+  materialize,
+} from 'rxjs/operators';
+
+/** Represents config with named paratemeters for mapToAction */
+export interface MapToActionConfig<
+  InputAction extends Action,
+  OutputAction extends Action,
+  ErrorAction extends Action,
+  CompleteAction extends Action,
+  UnsubscribeAction extends Action
+> {
+  // Project function that produces the output actions in success cases
+  project: (action: InputAction, index: number) => Observable<OutputAction>;
```

Comment:
I don't think the project function would receive an action. It would receive whatever the result was of the applied operator, right? 

Recommend renaming `InputAction` to `Input` and remove the `extends Action` type restriction

Reply:
Hm... you are right, there could be many transformations to the action applied before `mapToAction`.

e.g.: `actions$.pipe(ofType(myAction), withLatestFrom(other$), mapToAction(...))`

Reply:
Done.